testsuite/tests/conftest.py

```diff
@@ -152,26 +152,30 @@ def openshift2(testconfig, skip_or_fail):
 @pytest.fixture(scope="session")
 def keycloak(request, testconfig, blame, skip_or_fail):
     """Keycloak OIDC Provider fixture"""
+    realm_name = blame("realm")
     try:
         testconfig.validators.validate(only="keycloak")
         cnf = testconfig["keycloak"]
         info = Keycloak(
             cnf["url"],
             cnf["username"],
             cnf["password"],
-            blame("realm"),
+            realm_name,
             "base-client",
             cnf["test_user"]["username"],
             cnf["test_user"]["password"],
         )
 
         info.commit()
-        return info
+        yield info
     except KeycloakAuthenticationError:
         return skip_or_fail("Unable to login into SSO, please check the credentials provided")
     except KeyError as exc:
         return skip_or_fail(f"SSO configuration item is missing: {exc}")
 
+    # Realm cleanup
+    info.master_realm.delete_realm(realm_name)
```

Comment:
`info.delete()` is sufficient

Reply:
Thx, I missed this method. Will refactor as suggested.

Reply:
Refactored!